packages/core-flows/src/fulfillment/steps/create-fulfillment.ts

```diff
@@ -0,0 +1,28 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { FulfillmentTypes, IFulfillmentModuleService } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+export const createFulfillmentStepId = "create-fulfillment"
+export const createFulfillmentStep = createStep(
+  createFulfillmentStepId,
+  async (data: FulfillmentTypes.CreateFulfillmentDTO, { container }) => {
+    const service = container.resolve<IFulfillmentModuleService>(
+      ModuleRegistrationName.FULFILLMENT
+    )
+
+    const fulfillment = await service.createFulfillment(data)
+
+    return new StepResponse(fulfillment, fulfillment.id)
+  },
+  async (id, { container }) => {
+    if (!id) {
+      return
+    }
+
+    const service = container.resolve<IFulfillmentModuleService>(
+      ModuleRegistrationName.FULFILLMENT
+    )
+
+    await service.cancelFulfillment(id)
```

Comment:
Wouldn't make sense to fully delete the fulfillment here, considering it's a compensating operation?

Reply:
From what I can understand, create can potentially call an external provider and cancel does the same. I think it would be beneficial to keep those as logged and not delete it.

Reply:
Ok make sense 👍